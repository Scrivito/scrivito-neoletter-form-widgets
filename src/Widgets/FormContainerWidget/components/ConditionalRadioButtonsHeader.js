import * as React from "react";
import * as Scrivito from "scrivito";
import { getFieldName } from "../utils/getFieldName";
import { SelectItem } from "./SelectComponent";
import { Mandatory } from "./MandatoryComponent";
import { HelpText } from "./HelpTextComponent";

export const RadioButtonsHeader = Scrivito.connect(
  ({ widget, onChangeSelected }) => {
    return (
      <div
        className={`mb-3 select-container ${
          Scrivito.isInPlaceEditingActive() ? "condition-wrapper" : ""
        }`}
      >
        <div className="select-title">
          <span className="text-super">{widget.get("title")}</span>
          {widget.get("required") && <Mandatory />}
          {widget.get("helpText") && <HelpText widget={widget} />}
        </div>
        <div className="row ">
          {widget.get("conditions").map((condition, index) => (
            <SelectItem
              selectionType={"radio"}
              name={getFieldName(widget)}
              onChange={onChangeSelected}
              value={condition.get("title")}
              id={condition.id()}
              required={widget.get("required")}
              key={index}
            />
          ))}
        </div>
      </div>
    );
  }
);
