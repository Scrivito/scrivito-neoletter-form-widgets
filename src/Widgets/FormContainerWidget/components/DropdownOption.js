import * as React from "react";

export const DropdownOption = ({ value, id }) => {
  return (
    <option value={value} id={id}>
      {value}
    </option>
  );
};
