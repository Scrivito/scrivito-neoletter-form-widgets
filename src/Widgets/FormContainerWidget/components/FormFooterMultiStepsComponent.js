import * as React from "react";
import * as Scrivito from "scrivito";

export const FormFooterMultiSteps = Scrivito.connect(
  ({
    widget,
    onPageChange,
    onSubmit,
    currentStep,
    isLastPage,
    stepsLength,
  }) => {
    return (
      <div className="form-buttons">
        <button
          className="btn btn-primary backward-button"
          onClick={() => onPageChange(false)}
          hidden={currentStep == 1 && !Scrivito.isInPlaceEditingActive()}
        >
          {widget.get("backwardButtonText")}
        </button>
        <div className="step-counter">{currentStep + " / " + stepsLength}</div>
        <button
          className="btn btn-primary forward-button"
          onClick={isLastPage ? onSubmit : () => onPageChange(true)}
        >
          {isLastPage
            ? widget.get("submitButtonText")
            : widget.get("forwardButtonText")}
        </button>
      </div>
    );
  }
);
