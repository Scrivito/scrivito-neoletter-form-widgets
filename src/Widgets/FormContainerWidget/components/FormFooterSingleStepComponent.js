import * as React from "react";
import * as Scrivito from "scrivito";

export const FormFooterSingleStep = Scrivito.connect(({ widget, onSubmit }) => {
  return (
    <div
      className={`${
        widget.get("singleSubmitButtonAlignment") === "block"
          ? ""
          : widget.get("singleSubmitButtonAlignment")
      }`}
    >
      <button
        className={`btn btn-primary${
          widget.get("singleSubmitButtonAlignment") === "block"
            ? " btn-block"
            : ""
        }`}
        onClick={onSubmit}
      >
        {widget.get("submitButtonText")}
      </button>
    </div>
  );
});
