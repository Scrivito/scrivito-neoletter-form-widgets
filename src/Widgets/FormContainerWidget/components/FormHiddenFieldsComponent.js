import * as React from "react";
import * as Scrivito from "scrivito";
import { getHistory } from "../../../config/history";

export const FormHiddenFields = ({ widget }) => {
  const [browserLocation, setBrowserLocation] = React.useState(null);
  React.useEffect(() => {
    const history = getHistory();
    console.log("And the history is....");
    console.log(history);
    if (!history) return;
    setBrowserLocation(locationToUrl(history.location));

    return history.listen(({ location }) =>
      setBrowserLocation(locationToUrl(location))
    );
  }, []);
  return (
    <>
      <input type="hidden" name="form_id" value={widget.get("formId")} />
      <input
        type="hidden"
        name="url"
        value={browserLocation || Scrivito.urlFor(widget.obj())}
      />
      <Scrivito.InPlaceEditingOff>
        <Scrivito.ContentTag content={widget} attribute="hiddenFields" />
      </Scrivito.InPlaceEditingOff>

      <HoneypotField />
    </>
  );
};

const HoneypotField = () => (
  <div aria-hidden="true" className="winnie-the-pooh">
    <input autoComplete="off" name="fax" tabIndex="-1" />
  </div>
);

function locationToUrl(location) {
  return `${window.location.origin}${location.pathname}${location.search}`;
}
