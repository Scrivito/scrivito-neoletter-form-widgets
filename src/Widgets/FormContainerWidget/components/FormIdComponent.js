import * as React from "react";
import * as Scrivito from "scrivito";
import { ContentProperty } from "../../../Components/ContentProperty";
import { getScrivitoFormWidgetConfig } from "../../../config/scrivitoConfig";

export const FormIdComponent = Scrivito.connect(({ widget }) => {
  const formSubmissionsHref = widget.get("formId")
    ? `https://edit.neoletter.com/i/${
        getScrivitoFormWidgetConfig().tenant
      }/forms/${widget.get("formId")}`
    : null;
  const uiContext = Scrivito.uiContext();
  if (!uiContext) return null;

  return (
    <div className={`scrivito_${uiContext.theme}`}>
      <div className="scrivito_detail_content">
        <div className="attribute_form_id_item">
          <ContentProperty
            content={widget}
            attribute="formId"
            title="Form ID"
            description="This ID identifies the form in Neoletter."
          />

          <a
            className={`scrivito_button ${
              formSubmissionsHref ? "scrivito_blue" : "scrivito_disabled"
            }`}
            href={formSubmissionsHref}
            target="_blank"
            rel="noreferrer"
          >
            View submissions
          </a>
        </div>
      </div>
    </div>
  );
});
