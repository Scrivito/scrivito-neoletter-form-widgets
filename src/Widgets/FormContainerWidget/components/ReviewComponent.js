import * as React from "react";
import Button from "react-bootstrap/Button";
import Modal from "react-bootstrap/Modal";

export const Review = ({ widget, reviewData, onHide }) => {
  const [show, setShow] = React.useState(false);

  React.useEffect(() => {
    setShow(true);
  }, [reviewData]);

  const handleClose = () => {
    setShow(false);
    onHide();
  };
  return (
    <Modal show={show} onHide={handleClose} size="lg">
      {widget.get("showReviewHeader") && (
        <Modal.Header style={{ alignSelf: "center" }}>
          <Modal.Title>{widget.get("reviewHeaderTitle")}</Modal.Title>
        </Modal.Header>
      )}
      <Modal.Body className="form-review">
        {reviewData.map((step, i) => (
          <div
            className="step-review-container"
            key={"step-review-container-" + i}
          >
            {widget.get("showStepsInReview") && (
              <h4 className="step-review">Step {i}</h4>
            )}
            {step.map((d, i) => (
              <ReviewItem key={"item-" + i} item={d} />
            ))}
          </div>
        ))}
      </Modal.Body>
      {widget.get("showReviewFooter") && (
        <Modal.Footer>
          <Button variant="primary" onClick={handleClose}>
            {widget.get("reviewCloseButtonText")}
          </Button>
        </Modal.Footer>
      )}
    </Modal>
  );
};

const ReviewItem = ({ item }) => {
  return (
    <div>
      <div className="review-item-title">
        <span>{item.title}</span>
      </div>
      <div className="review-item-value">{item.value}</div>
    </div>
  );
};
