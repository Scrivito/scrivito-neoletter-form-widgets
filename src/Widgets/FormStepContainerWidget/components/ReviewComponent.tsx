import * as React from "react";
import { Widget } from "scrivito";
import { ReviewContent, ReviewItemContent } from "../../../../types/types";

interface ReviewProps {
  widget: Widget;
  reviewContent: ReviewContent;
  onHide: () => void;
}
interface ReviewItemProps {
  item: ReviewItemContent;
}

export const Review: React.FC<ReviewProps> = ({
  widget,
  reviewContent,
  onHide
}) => {
  const [show, setShow] = React.useState(false);

  React.useEffect(() => {
    setShow(true);
  }, [reviewContent]);

  const handleClose = () => {
    setShow(false);
    onHide();
  };
  const handleOnClickContainer = (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>
  ) => {
    if (event.currentTarget == event.target) {
      handleClose();
    }
  };
  return (
    <div
      onClick={(e) => handleOnClickContainer(e)}
      hidden={!show}
      className="form-modal"
    >
      <div className="modal-dialog">
        <div className="modal-content">
          {widget.get("showReviewHeader") && (
            <div className="modal-header">
              <h4>{widget.get("reviewHeaderTitle") as string}</h4>
            </div>
          )}
          <div className="modal-body form-review">
            {reviewContent.map((steps, i) => (
              <div
                className="step-review-container"
                key={"step-review-container-" + i}
              >
                {widget.get("showStepsInReview") && (
                  <h4 className="step-review">Step {i}</h4>
                )}
                {steps.map((d, i) => (
                  <ReviewItem key={"item-" + i} item={d} />
                ))}
              </div>
            ))}
          </div>
          {widget.get("showReviewFooter") && (
            <div className="modal-footer">
              <button className="" onClick={handleClose}>
                {widget.get("reviewCloseButtonText") as string}
              </button>
            </div>
          )}
        </div>
      </div>
    </div>
  );
};

const ReviewItem: React.FC<ReviewItemProps> = ({ item }) => {
  return (
    <div>
      <div className="review-item-title">
        <span>{item.title}</span>
      </div>
      <div className="review-item-value">{item.value}</div>
    </div>
  );
};
