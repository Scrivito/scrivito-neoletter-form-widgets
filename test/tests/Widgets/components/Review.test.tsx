import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Review } from "../../../../src/Widgets/FormStepContainerWidget/components/ReviewComponent";
import { Widget } from "scrivito";
import { DummyWidget } from "../../../helpers/dummyWidget";

const onHideMock = jest.fn();
const reviewContent = [
  [
    { title: "Title 1", value: "Value 1" },
    { title: "Title 2", value: "Value 2" }
  ],
  [
    { title: "Title 3", value: "Value 3" },
    { title: "Title 4", value: "Value 4" }
  ]
];

const widget = new DummyWidget({
  showReviewHeader: true,
  reviewHeaderTitle: "Review Header Title",
  showStepsInReview: true,
  showReviewFooter: true,
  reviewCloseButtonText: "Close"
}) as unknown as Widget;

describe("Review", () => {
  it("renders the Review component with provided props, header and footer shown", () => {
    const { getByText } = render(
      <Review
        widget={widget}
        reviewContent={reviewContent}
        onHide={onHideMock}
      />
    );

    expect(getByText("Review Header Title")).toBeInTheDocument();
    expect(getByText("Step 0")).toBeInTheDocument();
    expect(getByText("Title 1")).toBeInTheDocument();
    expect(getByText("Value 1")).toBeInTheDocument();
    expect(getByText("Close")).toBeInTheDocument();

    fireEvent.click(getByText("Close"));
    expect(onHideMock).toHaveBeenCalledTimes(1);
  });

  it("renders the Review component without header", () => {
    widget.update({ showReviewHeader: false });

    const { queryByText } = render(
      <Review
        widget={widget}
        reviewContent={reviewContent}
        onHide={onHideMock}
      />
    );

    expect(queryByText("Review Header Title")).toBeNull();
  });

  it("renders the Review component without steps", () => {
    widget.update({ showStepsInReview: false });

    const { queryByText } = render(
      <Review
        widget={widget}
        reviewContent={reviewContent}
        onHide={onHideMock}
      />
    );

    expect(queryByText("Step 0")).toBeNull();
    expect(queryByText("Step 1")).toBeNull();
  });

  it("renders the Review component without footer", () => {
    widget.update({ showReviewFooter: false });
    const { queryByText } = render(
      <Review
        widget={widget}
        reviewContent={reviewContent}
        onHide={onHideMock}
      />
    );

    expect(queryByText("Close")).toBeNull();
  });

  it("renders correctly", () => {
    widget.update({ showReviewFooter: true, showStepsInReview: true });
    render(
      <Review
        widget={widget}
        reviewContent={reviewContent}
        onHide={onHideMock}
      />
    );
    expect(document.documentElement.innerHTML).toMatchSnapshot();
  });
});
