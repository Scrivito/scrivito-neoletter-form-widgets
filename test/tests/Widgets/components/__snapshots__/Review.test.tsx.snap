// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review renders correctly 1`] = `"<head></head><body style="overflow: hidden;"><div><div class="scrivito-neoletter-review-form-modal"><div class="review-modal-dialog"><div class="review-modal-content"><div class="review-modal-body form-review"><div class="step-review-container"><h4 class="step-review">Step 0</h4><div><div class="review-item-title"><span>Title 1</span></div><div class="review-item-value">Value 1</div></div><div><div class="review-item-title"><span>Title 2</span></div><div class="review-item-value">Value 2</div></div></div><div class="step-review-container"><h4 class="step-review">Step 1</h4><div><div class="review-item-title"><span>Title 3</span></div><div class="review-item-value">Value 3</div></div><div><div class="review-item-title"><span>Title 4</span></div><div class="review-item-value">Value 4</div></div></div></div><div class="review-modal-footer"><button class="">Close</button></div></div></div></div></div></body>"`;
